core/src/main/java/io/github/tcdl/msb/collector/CollectorManager.java

```diff
@@ -53,26 +53,21 @@ public void registerCollector(Collector collector) {
         String correlationId = collector.getRequestMessage().getCorrelationId();
         collectorsByCorrelationId.putIfAbsent(correlationId, collector);
 
-        synchronized (this) {
-            if (!isSubscribed) {
-                channelManager.subscribeForResponses(topic, this);
-                isSubscribed = true;
+        if(!isSubscribed) {
+            synchronized (this) {
+                if (!isSubscribed) {
+                    channelManager.subscribeForResponses(topic, this);
+                    isSubscribed = true;
+                }
             }
         }
     }
 
     /**
-     * Remove this collector from collector's map, if it is present. If map is empty (no more collectors await on consumer topic) unsubscribe from consumer.
+     * Remove this collector from collector's map, if it is present.
      */
     public void unregisterCollector(Collector collector) {
         collectorsByCorrelationId.remove(collector.getRequestMessage().getCorrelationId());
-
-        synchronized (this) {
-            if (collectorsByCorrelationId.isEmpty() && isSubscribed) {
-                channelManager.unsubscribe(topic);
-                isSubscribed = false;
-            }
-        }
```

Comment:
As far as responses topic is static, I don't see a need in this "unsubscripe" call that has a significant negative performance impact (a lot of subscribe/unsubscribe cycles). I'll create msb-node change request regarding it.